File: 백준/Gold/1484. 다이어트/다이어트.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.ArrayList;
import java.util.List;

public class Main {
    public static void main(String[] args) throws IOException {

        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        int g = Integer.parseInt(br.readLine());

        long s = 1, e = 1;
        List<Long> ans = new ArrayList<>();

        while (e < g) {

            long diff = e * e - s * s;

            if (diff == g) {
                ans.add(e);
                e++;
            } else if (diff < g) {
                e++;
            } else {
                s++;
            }
        }

        if (ans.isEmpty()) {
            System.out.println(-1);
        } else {
            for (long a : ans) {
                System.out.println(a);
            }
        }
    }
}

File: 백준/Gold/1484. 다이어트/다이어트.py
g = int(input())

s, e = 1, 1
ans = []

while e < g:

    diff = e * e - s * s

    if diff == g:
        ans.append(e)
        e += 1
    elif diff < g:
        e += 1
    else:
        s += 1

if not ans:
    print(-1)
else:
    for a in ans:
        print(a)